File: examples/src/Shapes/AnimationTargetShapes.cs
using System
using System.IO
import aspose.slides as slides
using Aspose.slides.Animation
using Aspose.slides.DOM.Ole
using Aspose.slides.Export

/*
This sample demonstrates the output of information for all animated shapes in the main sequence for all slides in a presentation.
*/

namespace Aspose.slides.Examples.CSharp.shapes
{
    public class AnimationTargetShapes
    {
        public static void Run()
        {
            # Path to source presentation
            presentationFileName = Path.Combine(RunExamples.GetDataDir_Shapes(), "AnimationShapesExample.pptx")

            using (Presentation pres = new Presentation(presentationFileName))
            {
                foreach (slide in pres.Slides)
                {
                    foreach (IEffect effect in slide.timeline.main_sequence)
                    {
                        print(effect.type + " animation effect is set to shape#" +
                                          effect.TargetShape.UniqueId +
                                          " on slide#" + slide.SlideNumber)
                    }
                }
            }
        }
    }
}

File: examples/src/Shapes/AnimationTargetShapes.py
import aspose.slides as slides

# Path to source presentation
dataDir = "./examples/data/"
outDir = "./examples/out/"

with slides.Presentation(dataDir + "shapes_animation_example.pptx") as pres:
    for slide in pres.slides:
        for effect in slide.timeline.main_sequence:
            print("{0} animation effect is set to shape#{1} on slide#{2}".format( 
                effect.type, effect.target_shape.unique_id, slide.slide_number))